### ISSUE TITLE
Avoid using a "N/A" risk evaluation in the Transaction details

### ISSUE THREAD

**Issue by csmcneill** [Automattic/woocommerce-payments#8250]
### Describe the bug

When a payment is made via a BNPL payment method, there is no risk evaluation made. However, what happens as a result is that there is a value of `N/A` for the risk evaluation in the transaction details at **Payments > Transactions** without any further information.

### To Reproduce

1. Purchase a product via a BNPL payment method.
2. View that transaction's details via **Payments > Transactions**.
3. Note the `RISK EVALUATION` is `N/A`.

### Actual behavior

There is no indication as to why a risk evaluation was undertaken for this payment. While this was highlighted during an evaluation of BNPL payment methods, this could also apply to other instances where there was no risk evaluation made.

### Screenshots

![image](https://github.com/Automattic/woocommerce-payments/assets/82999806/e84a9907-460d-440c-b1cb-341614b6ece2)

### Expected behavior

When it’s expected for no risk evaluation to occur for a transaction, we would want to consider either:
1. Conditionally hiding this.
2. Showing the `N/A` but including a tooltip that informs the merchant why no risk evaluation was performed.

### Additional context

Surfaced in pbUcTB-4el-p2#risk-evaluation-n-a

**Comment by pierorocca** [comment]
Marking as an enhancement request. @elizaan36 @nikkivias BNPL payment methods have no fraud chargebacks. It's up to the BNPL providers to approve or decline and they assume all risk. So there's never a fraud assessment on our side. While we know that's what N/A means, it may be ambiguous for a merchant.

Do you feel N/A is sufficient? Would a tooltip with a legend be useful to explain what each value means? Thanks for the help

**Comment by elizaan36** [comment]
I don't think N/A is the best way to communicate that no risk evaluation was completed. I'd lean towards hiding the risk evaluation section from BNPL transactions (including on order details). I think any explanation would just generate more questions and this type of information could more easily be explained in the docs.

**Comment by pierorocca** [comment]
@elizaan36 I'm in the process of stress testing WooPay. For this 3D secure transaction, when I look at the Transaction details, I don't understand what Risk Evaluation 'Normal' means. Searching for "risk evaluation" in the docs yields no result.

![image](https://github.com/Automattic/woocommerce-payments/assets/109928953/278e336f-872c-4444-96aa-c1fee0bfdd4c)

![image](https://github.com/Automattic/woocommerce-payments/assets/109928953/8af770a9-e5c2-4ed9-8895-a1a6ab6e24b6)

**Comment by elizaan36** [comment]
That's a great point, @pierorocca. The [Fraud tools documentation](https://woo.com/document/woopayments/fraud-and-disputes/fraud-protection/) is quite detailed but the user may not know how to find it if they search for the terminology used in the admin.

@csmcneill What are your thoughts on changing the language in the admin vs adding the "Risk evaluation" term in the docs? For example, I think the metabox in the order details should be labeled "Risk evaluation" for consistency. Currently it's "Fraud & Risk".

<img width="604" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/5121465/a7596eda-f938-42bb-8bf6-0f3792d4242b">

**Comment by csmcneill** [comment]
> I think the metabox in the order details should be labeled "Risk evaluation" for consistency. Currently it's "Fraud & Risk".

~~100% agree here, as I think consistency is a vital component of delight :)~~

It seems as though the `Risk evaluation` section of the Transactions page and the `Fraud & Risk` metabox in the order details page are quite different. The `Risk evaluation` is pulled from Stripe, which also explains why there may be a `N/A` for BNPL transactions.

If an order is blocked by our FRT, there's no risk evaluation performed, and the value on the transactions page is `–`:

![CleanShot 2024-03-06 at 15 21 21](https://github.com/Automattic/woocommerce-payments/assets/82999806/0e4a5684-fbd8-456e-b680-324476a8a0d4)

> What are your thoughts on changing the language in the admin vs adding the "Risk evaluation" term in the docs?

I'll work on adding some information about `Risk evaluation` to our docs. It's likely that it will fall under https://woo.com/document/woopayments/fraud-and-disputes/fraud-protection/

**Comment by pierorocca** [comment]
Nice analysis @csmcneill. 

> If an order is blocked by our FRT, there's no risk evaluation performed, and the value on the transactions page is –:

I suspect this will vary based on the fraud check, by who and when it's being checked. Roughly the fraud and risk steps done in a typical transaction are:

1. WooPayments pre-auth fraud & risk checks & SIFT scoring
2. Stripe pre-auth fraud & risk checks (Radar)
3. Issuer fraud checks and auth response (decline or approve + useful metadata e.g. approved but zip didn't match). This could also include a 3DS check within the auth flow where the user is forced to validate identity.
5. Stripe post-auth fraud & risk checks (Radar - they may decide to decline an issuer approved transaction to limit risk)
6. WooPayments post-auth fraud & risk checks

2-4 is combined in the auth response message we get from Stripe.

1 & 5 are related to the new WooPayments Advanced fraud rules. 
- AVS Mismatch - post-auth
- International IP Address - pre-auth
- IP Address Mismatch - pre-auth
- Address Mismatch - pre-auth
- Purchase Price Threshold - pre-auth
- Order Items Threshold - pre-auth
- CVC Verification - post-auth

If you simulate an AVS mismatch I think you'll get both a risk evaluation and an AVS mismatch based on the Advanced fraud rule firing post-auth. How these scenarios combine if at all to populate "Risk Evaluation" I'd recommend a fresh look and have the owning PM weigh in.


Line1 check fails | 4000000000000028 | The address line 1 check fails.The payment succeeds unless you block it with a custom Radar rule.
-- | -- | --

![image](https://github.com/Automattic/woocommerce-payments/assets/109928953/590a4e90-ee17-4996-86dc-181b6a1e8ea4)

**Comment by pierorocca** [comment]
Not included above are other checks done with Sift or other fraud squad tools prior to initiating the authorization request.

**Comment by csmcneill** [comment]
My tests suggest that we model the `Risk evaluation` section based on the Stripe [risk outcomes](https://docs.stripe.com/radar/risk-evaluation#risk-outcomes). I did a bit of digging and found some evidence in the plugin code that supports this hypothesis:

https://github.com/Automattic/woocommerce-payments/blob/ca8c7c0d04400b6e4753e6baa919209e88752154/client/transactions/strings.ts#L38-L43

https://github.com/Automattic/woocommerce-payments/blob/ca8c7c0d04400b6e4753e6baa919209e88752154/client/components/risk-level/index.tsx#L14-L16

https://github.com/Automattic/woocommerce-payments/blob/ca8c7c0d04400b6e4753e6baa919209e88752154/client/components/risk-level/strings.ts#L8-L14

It's possible that I could be missing something. However, I do want to note that I'm unable to get this section of the payment details page to display anything other than `Normal`, `Elevated`, or `N/A`.

**Comment by csmcneill** [comment]
> If you simulate an AVS mismatch I think you'll get both a risk evaluation and an AVS mismatch based on the Advanced fraud rule firing post-auth.

I've never been able to get AVS mismatch failures to work in test mode, so I tried to do this on my live account with a live payment method.

The payment was authorized as expected, but failed due to the AVS mismatch. However, as a result, an order was created in wp-admin, but there is no associated transaction in **Payments > Transactions**.

Since there's no transaction, there's no risk evaluation. What's more interesting is that the FRT metabox suggests no action was taken here, even though it was blocked by an adjustment in my FRT settings:

![CleanShot 2024-03-06 at 18 36 19](https://github.com/Automattic/woocommerce-payments/assets/82999806/28119f0a-76b6-41a9-a325-78e1fe44a51b)

I'll open up a separate issue about mismatch between the FRT settings and the metabox information :)

**Comment by pierorocca** [comment]
Awesome super helpful @csmcneill. I found [Stripe's docs for Risk Evaluation](https://docs.stripe.com/radar/risk-evaluation). 

I don't know enough about the WooPayments advanced fraud implementation and SIFT scoring to know how they interplay with each other, and then how that's all packaged and communicated to the merchant. For example if Radar comes back with a Normal score but an advanced fraud rule triggers and declines the transaction, are the order notes, F&R notes, and risk evaluation conveying a clear or a muddied picture?

**Comment by pierorocca** [comment]
@csmcneill last week I also noticed during testing that anytime an order is blocked due to a fraud rule like IP mismatch which happens before authorization, an order is created in a “Pending payment” state. I pinged my team and Sourav provided a first response and I've followed up with another. I'm waiting on more feedback.

**Comment by pierorocca** [comment]
Despite it being a "Pending payment" order, which to me is odd and should be either Failed or Sourav is suggesting "On Hold", I do now see the blocked transaction. 

![image](https://github.com/Automattic/woocommerce-payments/assets/109928953/cd1ab824-6461-47f3-9219-1238b512a8ae)

**Comment by tpaksu** [comment]
Hello all, I'm triaging issues assigned to us, and I wanted to check if this task has a final decision about what should be done. This issue contains many good investigations and info, but I couldn't spot a decision about how we should replace `N/A` on the screen, which may indicate Stripe didn't return a risk evaluation result to us, which can be many things, like Radar isn't active on the account. 

Wild guess, didn't check the code, but `-` can mean the process didn't reach to a point where a risk evaluation can be done on the transaction, like FRT blocking the intent creation before it reaches Stripe. AFAIK, the transaction data on the "blocked" tab isn't showing data received from Stripe, it is mimicking the data as it was received from Stripe. We use what we have to build that screen, to look like the other "failed/succeeded" records screen. Right @eduardoumpierre?

**Comment by eduardoumpierre** [comment]
> AFAIK, the transaction data on the "blocked" tab isn't showing data received from Stripe, it is mimicking the data as it was received from Stripe. We use what we have to build that screen, to look like the other "failed/succeeded" records screen.

Hey! That's correct. Since the checkout is blocked before reaching Stripe, we generate a charge based on the [data available in the order](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/admin/class-wc-rest-payments-charges-controller.php#L89-L120). We set `charge.outcome` to `null` [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/admin/class-wc-rest-payments-charges-controller.php#L104), so a "-" is rendered in this case (and any other case where `charge.outcome` is unavailable).

Here is where the Risk Evaluation item in the summary page [is generated](https://github.com/Automattic/woocommerce-payments/blob/develop/client/payment-details/summary/index.tsx#L160-L165).

**Comment by dwainm** [comment]
@csmcneill @pierorocca . I'm leaning toward @elizaan36 's comment:
 https://github.com/Automattic/woocommerce-payments/issues/8250#issuecomment-1973900731

If not, could we close the issue and re-open when we have the right direction?

**Comment by pierorocca** [comment]
For non card payments where there's no Stripe risk assessment hiding it would make sense since it will always be N/A and that would trigger more questions than value it provides.

**Comment by dwainm** [comment]
Thanks for confirming @pierorocca. Having N/A isn't very helpful, even for card payments. We should only show it if we have data. "Unknown" or "not assessed", irrespective of payment method, should not be displayed, slightly simplifying the interface code.

**Comment by vbelolapotkov** [comment]
@csmcneill is there any real confused merchant by this? I don't see any feedback or evidence of it on original post.

**Comment by vbelolapotkov** [comment]
cc @elizaan36 in case you have an additional input on that ^^

**Comment by csmcneill** [comment]
@vbelolapotkov That's correct; AFAIK, there haven't been merchant reports of this being a pain point.

**Comment by elizaan36** [comment]
I don't think it's a critical issue based on merchant feedback but we can aim to prevent potential confusion where we can.

**Comment by vbelolapotkov** [comment]
Ok, I don't mind iterating on UI/UX enhancements, but let's close this issue and open a new one when there is a clear proposal for improvement.
